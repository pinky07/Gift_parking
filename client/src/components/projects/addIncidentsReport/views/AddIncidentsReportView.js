import React from 'react';

import Heading from 'grommet/components/Heading';
import Form from 'grommet/components/Form';
import FormFields from 'grommet/components/FormFields';
import FormField from 'grommet/components/FormField';
import TextInput from 'grommet/components/TextInput';
import NumberInput from 'grommet/components/NumberInput';
import Box from 'grommet/components/Box';
import Footer from 'grommet/components/Footer';
import Button from 'grommet/components/Button';
import DateTime from 'grommet/components/DateTime';
import Layer from 'grommet/components/Layer';

const AddIncidentsReportView = ({ viewModel }) => {
  const { errors } = viewModel;
  const { formCallbacks } = viewModel;
  const { projectName } = viewModel;
  const { reportDate } = viewModel;
  const { totalIncidents } = viewModel;
  const { rationale } = viewModel;

  return (
    <Layer align='center' closer={true} onClose={formCallbacks.onClose}>

      <Box pad={{ vertical: 'large', horizontal: 'small' }}>

        <Heading>Add an incidents report</Heading>

        <Form compact={false}>
          <FormFields>
            <fieldset>

              <FormField label='Report date' htmlFor="reportDateInput" error={errors.reportDate}>
                <DateTime id='reportDateInput' format='YYYY-MM-DD' value={reportDate} onChange={formCallbacks.onReportDateChange} />
              </FormField>

              <FormField label='Total incidents' htmlFor='totalIncidentsInputId' error={errors.totalIncidents}>
                <NumberInput id='totalIncidentsInputId' step={1} min={1} max={100} value={totalIncidents} onChange={formCallbacks.onTotalIncidentsChange} />
              </FormField>

              <FormField label='Rationale' htmlFor="rationaleInputId" error={errors.rationale}>
                <textarea rows={4} id='rationaleInputId' value={rationale} onChange={formCallbacks.onRationaleChange} />
              </FormField>

            </fieldset>
          </FormFields>

          <Footer pad={{ "vertical": "medium" }}>
            <Box>
              <Button label='Add' type='submit' primary={true} onClick={formCallbacks.onClick} />
            </Box>
            <Box pad={{ horizontal: 'small' }}>
              <Button label='Cancel' secondary={true} onClick={formCallbacks.onClose} />
            </Box>
          </Footer>
        </Form>
      </Box>
    </Layer>
  );
};

export default AddIncidentsReportView;