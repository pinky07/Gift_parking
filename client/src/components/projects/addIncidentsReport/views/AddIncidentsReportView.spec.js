import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import AddIncidentsReportView from './AddIncidentsReportView';

describe('ReportIncidentsView.spec', () => {

    it('should render the correct data values and validation errors', () => {
        const viewModel = getViewModelWithErrors();
        const renderer = ShallowRenderer.createRenderer();
        const actual = renderer.render(<AddIncidentsReportView viewModel={viewModel} />);

        expect(actual).toMatchSnapshot();
    });

    function getViewModelWithErrors() {
        return {
            projectName: 'Exceptional Project',
            reportDate: '2017-01-02',
            totalIncidents: 2,
            rationale: 'Some issues',
            errors: {
                reportDate: 'error reportDate',
                totalIncidents: 'error totalIncidents',
                rationale: 'error rationale',
            },
            formCallbacks: {
                onReportDateChange: jest.fn(),
                onTotalIncidentsChange: jest.fn(),
                onRationaleChange: jest.fn(),
                onClick: jest.fn(),
                onClose: jest.fn()
            }
        };
    }
});