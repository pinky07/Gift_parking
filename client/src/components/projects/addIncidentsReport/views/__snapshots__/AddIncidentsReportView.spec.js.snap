// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportIncidentsView.spec should render the correct data values and validation errors 1`] = `
<Layer
  align="center"
  closer={true}
  onClose={[Function]}
>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad={
      Object {
        "horizontal": "small",
        "vertical": "large",
      }
    }
    responsive={true}
    tag="div"
  >
    <Heading
      tag="h1"
    >
      Add an incidents report
    </Heading>
    <Form
      compact={false}
      fill={false}
      pad="none"
    >
      <FormFields>
        <fieldset>
          <FormField
            error="error reportDate"
            htmlFor="reportDateInput"
            label="Report date"
            size="medium"
            strong={false}
          >
            <DateTime
              format="YYYY-MM-DD"
              id="reportDateInput"
              onChange={[Function]}
              step={1}
              value="2017-01-02"
            />
          </FormField>
          <FormField
            error="error totalIncidents"
            htmlFor="totalIncidentsInputId"
            label="Total incidents"
            size="medium"
            strong={false}
          >
            <NumberInput
              id="totalIncidentsInputId"
              max={100}
              min={1}
              onChange={[Function]}
              step={1}
              value={2}
            />
          </FormField>
          <FormField
            error="error rationale"
            htmlFor="rationaleInputId"
            label="Rationale"
            size="medium"
            strong={false}
          >
            <textarea
              id="rationaleInputId"
              onChange={[Function]}
              rows={4}
              value="Some issues"
            />
          </FormField>
        </fieldset>
      </FormFields>
      <Footer
        align="center"
        direction="row"
        pad={
          Object {
            "vertical": "medium",
          }
        }
        responsive={false}
      >
        <Box
          announce={false}
          direction="column"
          focusable={true}
          pad="none"
          responsive={true}
          tag="div"
        >
          <Button
            label="Add"
            method="push"
            onClick={[Function]}
            primary={true}
            type="submit"
          />
        </Box>
        <Box
          announce={false}
          direction="column"
          focusable={true}
          pad={
            Object {
              "horizontal": "small",
            }
          }
          responsive={true}
          tag="div"
        >
          <Button
            label="Cancel"
            method="push"
            onClick={[Function]}
            secondary={true}
            type="button"
          />
        </Box>
      </Footer>
    </Form>
  </Box>
</Layer>
`;
